baybe/surrogates/base.py

```diff
@@ -62,8 +59,33 @@ class _NoTransform(Enum):
 """Sentinel to indicate the absence of a transform where `None` is ambiguous."""
 
 
+class SurrogateProtocol(Protocol):
+    """Type protocol specifying the interface surrogate models need to implement."""
+
+    def fit(
+        self,
+        searchspace: SearchSpace,
+        objective: Objective,
+        measurements: pd.DataFrame,
+    ) -> None:
+        """Fit the surrogate to training data in a given modelling context.
+
+        For details on the expected method arguments, see
+        :meth:`baybe.recommenders.base.RecommenderProtocol`.
+        """
+
+    def to_botorch(self) -> Model:
+        """Create the botorch-ready representation of the fitted model.
+
+        The :class:`botorch.models.model.Model` created by this method needs to be
+        configured such that it can be called with candidate points in **computational
+        representation**, that is, input of the form as obtained via
+        :meth:`baybe.searchspace.core.SearchSpace.transform`.
+        """
+
+
 @define
-class Surrogate(ABC, SerialMixin):
+class Surrogate(ABC, SurrogateProtocol, SerialMixin):
```

Comment:
But it is two separate concerns, really. `ABC` simply says that the class is abstract, i.e. you can't instantiate it, because it contains abstract methods. `Protocol`s fulfill a similar purpose when in comes to abstraction in that they impose a certain interface, but the difference is that, strictly speaking, no inheritance is required (see "nominal subtyping" vs "structural subtyping", eg in the [mypy docs](https://mypy.readthedocs.io/en/stable/protocols.html#protocols-and-structural-subtyping)). 

The point is: we explicitly indicate that our surrogate class fulfills the contracts imposed by `SurrogateProtocol` by subclassing from it (which is not necessary) but it **still is** an abstract class.